csemlib/utils.py

```diff
@@ -30,15 +30,17 @@ def cart2sph(x, y, z):
     :return: colatitude, longitude, and radius
     """
 
-    r = np.math.sqrt(x ** 2 + y ** 2 + z ** 2)
+    x, y, z = np.asarray(x), np.asarray(y), np.asarray(z)
+    r = np.sqrt(x ** 2 + y ** 2 + z ** 2)
+
     # Break obviously if radius is zero. In that case we're just
     # at the center of the Earth.
-    try:
-        c = np.math.acos(z / r)
-    except ZeroDivisionError:
-        return 0, 0, 0
-    l = np.math.atan2(y, x)
+    with np.errstate(invalid='ignore'):
```

Comment:
This handles floating point errors such as dividing by zero at the core. 
Dividing by zero in this function will now return a nan without warning, which is later replaced by zero